gomodcmd.go

```diff
@@ -56,29 +57,31 @@ func listModules(modules ...string) (mods map[string]*listModule, err error) {
 	return mods, nil
 }
 
-func goModInfo() (*modfile.File, error) {
-	// Get info on the current module so that we can find the
-	// path to the current go.mod file.
-	mods, err := listModules()
+// goModInfo returns the main module's root directory
+// and the parsed contents of the go.mod file.
+func goModInfo() (string, *modfile.File, error) {
+	goModPath, err := findGoMod(cwd)
 	if err != nil {
-		return nil, errors.Wrap(err)
-	}
-	if len(mods) != 1 {
-		return nil, errors.Newf("unexpected module list count (want 1 got %d)", len(mods))
-	}
-	var goModPath string
-	for _, m := range mods {
-		goModPath = m.GoMod
+		return "", nil, errors.Notef(err, nil, "cannot find main module")
 	}
+	rootDir := filepath.Dir(goModPath)
 	data, err := ioutil.ReadFile(goModPath)
 	if err != nil {
-		return nil, errors.Notef(err, nil, "cannot read current go.mod file")
+		return "", nil, errors.Notef(err, nil, "cannot read main go.mod file")
 	}
 	modf, err := modfile.Parse(goModPath, data, nil)
 	if err != nil {
-		return nil, errors.Wrap(err)
+		return "", nil, errors.Wrap(err)
+	}
+	return rootDir, modf, nil
+}
+
+func findGoMod(dir string) (string, error) {
+	out, err := runCmd(dir, "go", "env", "GOMOD")
```

Comment:
If run when not inside a module directory, this appears to return `/dev/null`, and does not report an error.

Reply:
Excellent point, thanks. Fixed.